scripts/test.sh

```diff
@@ -44,16 +48,33 @@ stop_server() {
   fi
 }
 
+startedSimulator=false
+log_temp=
+test_temp_dir=
 cleanup() {
   # Kill started object server
-  stop_server
-
+  stop_server || true
+  
+  # Give the simulator a chance to go down gracefully
+  if $startedSimulator; then
+    osascript -e 'tell app "Simulator" to quit without saving' || true
+    sleep 0.25 # otherwise the pkill following will get it
```

Comment:
Using `osascript` is just to give it a chance to go down nicely, reducing the amount of error clutter in stderr/stdout (before this was always giving nasty-looking messages). So the .25 secs is more of a grace period before the following `pkill -P $$` (it is now a child of ours). Additionally, since we are in `cleanup` I would rather not wait infinitely for something that is not going to go down from a `SIGHUP`. That sort of behavior is more appropriate in startup than shutdown (i.e.: line 164).